### DISCUSSION TITLE
Keeping the bot online

### DISCUSSION THREAD

**Discussion by Frwf** [jagrosh/MusicBot#704] [Questions] [answered]
So, i hosted the music bot and it worked properly. BUT. When i close the .jar (so the java) the bot is off and doesn't react anymore. Can i fix this in the config.txt or can you send me a different .jar? Thank you!

Greetings, Ducky.

**Comment by SplitPixl** [comment, marked as answer]
You must be running the jar for the bot to run.

**Comment by minecraftchest1** [comment]
If you are running under Linux, you can create an SystemD script (ping me for help with that), if you are under Windows, you can do it with task scheduler.

**Reply by ipetko96** [reply]
@minecraftchest1 Hi, please can you help me with systemd script? I set it up with the help of some tutorials, but it don't work.
I am missing something. Linux is not new to me, but systemd script are.
My discord: uptlak#9751
Thank you

**Reply by hermitoff** [reply]
@minecraftchest1 Hi, me too would like to learn how to use SystemD on linux please. ^-^
Thank you !

**Reply by ipetko96** [reply]
nvm, I figure it out, now it works as it should :)
@hermitoff If you need help, ping me on discord.

**Reply by minecraftchest1** [reply]
Never just say I figured it out. Please share how you did so to help others with the same issue.

**Reply by minecraftchest1** [reply]
I am somewhat knowledgeable in SystemD.

**Reply by ipetko96** [reply]
Guys don't worry, I am writing little guide for you all, but last days I was very busy at work.

**Reply by minecraftchest1** [reply]
put in `/usr/lib/systemd/system/JMusicBot.service` and edit to match your setup.
```
#Sample SystemD config file.
#See https://wiki.archlinux.org/title/Systemd
#and systemd.service(5) for more detials.

[Unit]
Description=JMusicBot for discord
#Waits for the network before starting up.
After=network.target
Requires=network.target

[Service]
Type=simple
#Command to run. It is recomended to use the full path.
ExecStart=/usr/bin/java -Xms512M -Xmx512M -Dnogui=true -jar /opt/JMusicBot/JMusicBot-0.3.4.jar
#How long after starting the service to wait for a
#sucessful start notifaction before stoping the sercive
#and marking it as failed.
TimeoutStartSec=3600
#Wether to restart the service if it fails.
Restart=always
#How long after the service fails to restart it.
RestartSec=10
#The path to the directory to change to before starting
#the service.
WorkingDirectory=/opt/JMusicBot
#The user and group to start the service as.
User=bot
Group=bot
#I don't know.

[Install]
#The target to install the service into.
WantedBy=multi-user.target
```

**Comment by qomiter** [comment]
you can start it in the background with nohup or run it with "screen" and it will stay running when you close the terminal.

**Comment by ipetko96** [comment]
I made bash script that will create systemctl service for the Bot.
You have to [configure the Bot](https://github.com/jagrosh/MusicBot/wiki/Setup) before this script.
Just paste it in the console and follow instructions.

#### Copy everything at once, it is one command!
`cd ; curl https://gist.githubusercontent.com/ipetko96/46f14f79391ed2405f8ad3f117d02311/raw/39e7c419a3814cf2813b5c102926060b557eec99/createJMusicBotService.sh --output createJMusicBotService.sh ; sudo bash createJMusicBotService.sh ; rm createJMusicBotService.sh`

It will automatically:
- download [bash script](https://gist.github.com/ipetko96/46f14f79391ed2405f8ad3f117d02311)
- execute as root
- create systemctl service
- set correct permissions
- enable the service
- clean after itself.

Now, after every system reboot, Bot will automatically start.
###### @jagrosh maybe add it to official wiki? Or at least link to this discussion?

**Comment by battlemarck** [comment]
you can host it on repl.it website with uptimerobot
